docs/source/rfc/0001-data-authorization-middleware.md

```diff
@@ -0,0 +1,628 @@
+- Office-by-office migration capability
+- Emergency fallback to VPD-only mode
+- Data consistency validation tools
+
+### User Impact
+
+**End Users (Zero Impact):**
+
+- Same API endpoints and authentication
+- No client application changes required
+- Transparent authorization enhancement
+
+**Data Administrators:**
+
+- New web-based access management UI
+- Enhanced granular access controls
+- Real-time policy updates
+- Improved audit and reporting capabilities
+
+**Operations Team:**
+
+- New service to monitor and maintain
+- Enhanced security posture
+- Better troubleshooting capabilities
+- Standardized policy management
+
+### Local Development & Testing Plan
+
+1. **Local Environment Setup**
+
+   - Deploy using Docker/Podman Compose
+   - Configure all services (Database, API, Auth Service, OPA)
+   - Load sample timeseries test data
+   - Verify end-to-end connectivity
+
+2. **Timeseries Authorization Testing**
+
+   - Implement authorization for timeseries endpoints
+   - Test all 7 user personas with timeseries data
+   - Validate embargo rules and time-based constraints
+   - Performance benchmarking in local environment
+
+3. **Integration Validation**
+
+   - Test Authorization Helper Library with timeseries controllers
+   - Validate JWT pass-through and context headers
+   - Ensure VPD compatibility during transition
+   - Document integration patterns for future resources
+
+## Detailed Design
+
+### Implementation Strategy
+
+**Transparent Proxy Pattern:**
+
+- Authorization layer for `/cwms-data/timeseries/*` endpoints
+- Policy-based access control with intelligent caching
+- Single structured header for authorization context
+- Selective response filtering for embargo and persona rules
+
+**Java API Integration:**
+
+- Authorization Helper Library for timeseries controllers
+- Minimal changes to TimeSeriesController (2-3 lines)
+- Structured authorization context via single HTTP header
+- Focus on demonstrating pattern for future resources
+
+### Java API Integration Approach
+
+**Authorization Helper Library Pattern:**
+
+```java
+@RestController
+public class TimeSeriesController {
+    @Autowired
+    private CwmsAuthorizationHelper authHelper;
+
+    @GetMapping("/timeseries")
+    public ResponseEntity<List<TimeSeries>> getTimeSeries(
+            String office, Context ctx) {
+
+        QueryBuilder query = QueryBuilder.create().table("cwms_ts_data");
+        query = authHelper.applyAuthorization(query, ctx);  // Authorization applied
+
+        List<TimeSeries> results = timeSeriesDao.execute(query);
+        results = authHelper.applyResponseFiltering(results, ctx);  // Response filtering applied
+
+        return ResponseEntity.ok(results);
+    }
+}
+```
+
+**Integration Options:**
+
+1. **Dependency Injection**: `@Autowired` helper in controllers
+2. **Annotation-Based**: `@CwmsAuthorized` with AOP interceptors
+3. **Filter-Based**: Servlet filter for automatic processing
```

Comment:
Javalin has before handlers and a Context object:

https://github.com/USACE/cwms-data-api/blob/2c19ffccf88e3e5c2480760e3f138cfa5279763b/cwms-data-api/src/main/java/cwms/cda/ApiServlet.java#L353C17-L353C35

https://github.com/USACE/cwms-data-api/blob/develop/cwms-data-api/src/main/java/cwms/cda/security/Authenticator.java#L15

With the before handler functionally equivalent to the Filter Based approach.

I suspect a combination of work within an CdaAccessManager, a before handler, and the specific endpoints for any data specific issues will likely be what happens.

e.g. a POST call with no `x-cwms-auth-context` should fail before it before it even gets to the Time Series Controller.